lib/tapioca/dsl/compilers.rb

```diff
@@ -0,0 +1,24 @@
+# typed: strict
+# frozen_string_literal: true
+
+require "tapioca/rbi_ext/model"
+require "tapioca/dsl/helpers/param_helper"
+require "tapioca/dsl/pipeline"
+
+module Tapioca
+  module Dsl
+    module Compilers
+      DIR = T.let(
+        File.expand_path("compilers", __dir__),
+        String
+      )
+      NAMESPACES = T.let(
+        [
+          "#{name}::", # compilers in this namespace
+          "", # compilers that need to be fully namespaced
```

Comment:
Why is this needed?

Reply:
When resolving a compiler by name, we special case the compilers under the `Tapioca::Dsl::Compilers` namespace so that we can use their class names only. For all other compilers, they need to be fully qualified when passed by name. This array is the array of namespaces we search under given a compiler by name. Thus, this entry is the namespace that we expect 3rd party compilers to be found under. Maybe this should be `"::"`?

Reply:
^ Can we add this answer as comment? 🙏